### DISCUSSION TITLE
Referencing same cached data with multiple `queryKey`s

### DISCUSSION THREAD

**Discussion by derochedesign** [TanStack/query#7695] [Q&A] [answered]
Not sure if this is an anti-pattern, but I have two separate useQuery hooks that perform get requests, which will ultimately be capable of returning the exact same data.

Basically I want to be able to set a `queryKey` based on the results of a query, so that a different query can rely on that cache.

Some further context:

`Query A`: get the most recent item in a list based on oldest item
- in this case, the endpoint is given an old item (`id: 1`) and the backend will return the newest in that list (lets say `id: 5`)
- originally i set the `queryKey` to `['lists', 'recent', id]`

`Query B`: get list item by id
- here we just directly get an item based on the id
- `queryKey` is `['lists', id]`

`Query A` and `Query B` will in a lot of cases return the same data and I'd like to be able to reference the same cached data. I would expect something like: 

- once `Query A` fetches data, I update/create the `queryKey` for that data to be something like `['lists', result.data.id]`
- then if `Query B` tries to fetch, for example, `['lists', '5']`, it will get that from the cache that `Query A` created.

Hopefully this is clear; thanks for any guidance in advance.

**Comment by TkDodo** [comment, marked as answer]
take a look at seeing the cache: https://tkdodo.eu/blog/seeding-the-query-cache#seeding-details-from-lists

**Reply by derochedesign** [reply]
The push approach is exactly what i was looking for, thank you!

**Reply by derochedesign** [reply]
actually, in this case there would be two records in the cache of the same data right? rather than there being two different keys that reference the same data. the reason it matters to me (other than avoiding storing duplicate data) is that I'd like to be able to invalidate either key individually and have the result be that both keys (and of course the cached data) are invalidated. 

In my case, I don't have the data necessary to manually invalidate the second key when I invalidate the first one, since I don't know what args were used to fetch it originally. (for example, my first fetch could be based on an item `id:2` but return `id:5`, and the key would be `['lists', 'recent', '2']`. Then i manually create `['lists', '5']` with the result; but once i want to invalidate item `id:5`, I have no way to also invalidate the first `['lists', 'recent', '2']`) because I can't know what the original id used was)

if its not possible to have a many-to-one relationship between keys and cache then i'll have to reconfigure how some of my data flow works
